### ISSUE TITLE
Allow nway to be imported and used from python programs

### ISSUE THREAD

**Issue by minzastro** [JohannesBuchner/nway#7]
I would like to include nway as a module into my pipeline, to run cross-match from python, not from the command line. How can I do that? It would be nice to have a "function" implementation, not only command-line tool.

**Comment by JohannesBuchner** [comment]
I am working on this.

**Comment by JohannesBuchner** [comment]
Please have a look at the "api" branch. The nway-apitest.py script demonstrates how to use the nwaylib.nway_match function.

```
result = nwaylib.nway_match(
	[
	table_from_fits('doc/COSMOS_XMM.fits', area=2.0),
	table_from_fits('doc/COSMOS_OPTICAL.fits', poserr_value=0.1, area=2.0),
	],
	match_radius = 20, # in arcsec
	prior_completeness = 0.9,
)
```

This way, you will receive a pandas DataFrame.
The catalogues are passed as a list of dicts with all the information (`dict(name=table_name, ra=ra, dec=dec, error=poserr, area=area, mags=mags, maghists=maghists, magnames=magnames)`).

**Comment by minzastro** [comment]
Great! Thank you.

**Comment by JohannesBuchner** [comment]
commit 916217d74 refactored the code a bit, should be easier to understand now.

**Comment by JohannesBuchner** [comment]
This feature is implemented in the latest version (>4.0). The `nway-apitest.py` script demontrates how to use NWAY from python. Hope it works for you. If you can test and report back that would be appreciated.
